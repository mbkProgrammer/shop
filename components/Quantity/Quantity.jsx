/* eslint-disable no-param-reassign */
/* eslint-disable no-const-assign */
/* eslint-disable no-return-assign */
import { useState } from 'react';
import styled from '@emotion/styled';
import { css, cx } from '@emotion/css';
import { useTheme } from '@emotion/react';

const Quantity = ({ setItemNum, itemNum }) => {
  const theme = useTheme();

  const Quantity = styled.div`
  width: 75px;
  height: 61px;
  display: flex;
  margin: 5px;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  `;

  const QuantityBtn = styled.button`
  background: none;
  color: ${theme.colors.primary};
  font-size: 18px;
  font-weight: 400;
  border: none;
  padding: 0;
  width: 25px;
  height: 30.5px;
  margin: 0;
  transition: 0.3s;
  &:hover {
  background: ${theme.colors.primary};
  color: ${theme.colors.secondary};
  }
  &:disabled {
    color: #888;
  }
  &:disabled:hover {
    background: none;
    color: #666;
  }
  `;

  return (
    <Quantity>
      <div className={css`height: 50%; color: ${theme.colors.text}`}>
        Quantity
      </div>
      <div className={css`height: 50%; width: 100%; display: flex; justify-content: space-between; align-items: center; border: 1px solid ${theme.colors.primary}; cursor: pointer;`}>
        <QuantityBtn type="button" onClick={() => setItemNum(itemNum += 1)}>+</QuantityBtn>
        <div>{itemNum}</div>
        <QuantityBtn type="button" onClick={() => setItemNum(itemNum -= 1)} disabled={itemNum === 1}>-</QuantityBtn>
      </div>
    </Quantity>
  );
};

export default Quantity;
