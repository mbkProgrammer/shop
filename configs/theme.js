const theme = {
  colors: {
    primary: '#56B280',
    secondary: '#F5FBF8',
    text: '#222322',
    background: '#EAF6F0',
  },
  typography: {
    body: {
      fontWeight: '300',
      fontSize: '1rem',
    },
    h1: {
      fontWeight: '700',
      fontSize: '2.4rem',
    },
    h2: {
      fontWeight: '700',
      fontSize: '2.2rem',
    },
    h3: {
      fontWeight: '700',
      fontSize: '2rem',
    },
    h4: {
      fontWeight: '600',
      fontSize: '1.8rem',
    },
    h5: {
      fontWeight: '600',
      fontSize: '1.6rem',
    },
    h6: {
      fontWeight: '500',
      fontSize: '1.4rem',
    },
  },
};

export default theme;
