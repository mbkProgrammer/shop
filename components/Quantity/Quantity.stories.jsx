import React, { useState } from 'react';
import Quantity from './Quantity';

export default { title: 'components/Quantity' };
export const QuantityTest = () => {
  const [itemNum, setItemNum] = useState(1);
  return (<Quantity itemNum={itemNum} setItemNum={setItemNum} />);
};
