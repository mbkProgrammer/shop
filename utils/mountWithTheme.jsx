import { ThemeProvider } from '@emotion/react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../configs/Store';
import theme from '../configs/theme';

const mountWithTheme = (children) => renderer.create(
  <Provider store={store}>
    <ThemeProvider theme={theme}>
      {children}
    </ThemeProvider>
  </Provider>,
);

export default mountWithTheme;
