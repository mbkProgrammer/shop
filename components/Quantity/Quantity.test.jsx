import { ThemeProvider } from '@emotion/react';
import React, { useState } from 'react';
import renderer from 'react-test-renderer';
import {
  fireEvent, render, screen,
} from '@testing-library/react';
import Quantity from './Quantity';
import theme from '../../configs/theme';
import mountWithTheme from '../../utils/mountWithTheme';

test('Input smaple', () => {
  const component = mountWithTheme(
    <Quantity />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
