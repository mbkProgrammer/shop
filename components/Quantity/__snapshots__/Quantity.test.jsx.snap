// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input smaple 1`] = `
<div
  className="css-1i6o0lg"
>
  <div
    className="css-1uk9dl5"
  >
    <button
      className="css-hmo6fz"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <div />
    <button
      className="css-hmo6fz"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      -
    </button>
  </div>
</div>
`;
